builders/containerBuilder.go

```diff
@@ -0,0 +1,255 @@
+)
+
+// ContainerBuilder is a struct that can build containers
+type ContainerBuilder struct {
+	workloads.Container
+}
+
+// NewContainerBuilder creates a new container builder
+func NewContainerBuilder(nodeID string, flist string, capacity workloads.ContainerCapacity, networkConnection []workloads.NetworkConnection) *ContainerBuilder {
+	return &ContainerBuilder{
+		Container: workloads.Container{
+			NodeId:            nodeID,
+			Flist:             flist,
+			Capacity:          capacity,
+			NetworkConnection: networkConnection,
+		},
+	}
+}
+
+// LoadContainerBuilder loads a container builder based on a file path
+func LoadContainerBuilder(reader io.Reader) (*ContainerBuilder, error) {
+	container := workloads.Container{}
+
+	err := json.NewDecoder(reader).Decode(&container)
+	if err != nil {
+		return &ContainerBuilder{}, err
+	}
+
+	return &ContainerBuilder{Container: container}, nil
+}
+
+// Save saves the container builder to an IO.Writer
+func (c *ContainerBuilder) Save(writer io.Writer) error {
+	err := json.NewEncoder(writer).Encode(c.Container)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+// Build validates and encrypts the secret environment of the container
+func (c *ContainerBuilder) Build() (workloads.Container, error) {
+	// TODO check validity fields
+	if c.Container.Flist == "" {
+		return workloads.Container{}, fmt.Errorf("flist cannot be empty")
+	}
+
+	if c.Container.SecretEnvironment == nil {
+		c.Container.SecretEnvironment = make(map[string]string)
+	}
+
+	for k, value := range c.Container.Environment {
+		secret, err := encryptSecret(value, c.Container.NodeId)
+		if err != nil {
+			return workloads.Container{}, errors.Wrapf(err, "failed to encrypt env with key '%s'", k)
+		}
+		c.Container.SecretEnvironment[k] = secret
+	}
+	c.Container.Environment = make(map[string]string)
+	return c.Container, nil
+}
+
+// WithNodeID sets the node ID to the container
+func (c *ContainerBuilder) WithNodeID(nodeID string) *ContainerBuilder {
+	c.Container.NodeId = nodeID
+	return c
+}
+
+// WithFlist sets the flist to the container
+func (c *ContainerBuilder) WithFlist(flist string) *ContainerBuilder {
+	c.Container.Flist = flist
+	return c
+}
+
+// WithNetwork sets the networks to the container
+func (c *ContainerBuilder) WithNetwork(connections []workloads.NetworkConnection) *ContainerBuilder {
+	c.Container.NetworkConnection = connections
+	return c
+}
+
+// WithInteractive sets the interactive flag to the container
+func (c *ContainerBuilder) WithInteractive(interactive bool) *ContainerBuilder {
+	c.Container.Interactive = interactive
+	return c
+}
+
+// WithHubURL sets the hub url to the container
+func (c *ContainerBuilder) WithHubURL(url string) *ContainerBuilder {
+	c.Container.HubUrl = url
+	return c
+}
+
+// WithEnvs sets the environments to the container
+func (c *ContainerBuilder) WithEnvs(envs []string) (*ContainerBuilder, error) {
+	environments, err := splitEnvs(envs)
```

Comment:
@muhamadazmy do we want to return errors from some `With..` methods? Because in the tfuser client for example string are passed and then we need to parse this to the actual objects..

Reply:
No, with should take valid input always, because it's not made for cmd line parsing. the cmd line tool will still have to do the argument parsting. Hence WithEnv should accept a map[string]string.